accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,165 @@
+
+To facilitate this, we will package source generators inside [reference packs](https://github.com/dotnet/designs/blob/main/accepted/2019/targeting-packs-and-runtime-packs.md).  We will use the standard [NuGet conventions for describing analyzers](https://docs.microsoft.com/en-us/nuget/guides/analyzers-conventions) with respect to the location in the package, however the SDK will not probe this path.  All source generators and their metadata will be listed in FrameworkList.xml, similar to references.  Analyzers will be added as `File` elements with `Type="Analyzer"` (constrasting to `Type="Managed"` which is currently used for references) and optionally `Language="cs|vb"`.  Omitting the `Language` attribute means the analyzer applies to all languages.  The .NET SDK will be responsible for locating the appropriate source generators from the refpack, based on the project's `$(Language)` and creating `@(Analyzer)` items as part of the `ResolveTargetingPackAssets` task.  These items will be conflict-resolved with other `@(Analyzer)` sources as described in the following section.  When viewed from the IDE it should be clear that the source of the `Analyzer` is the framework reference, similarly to how it is clear that the source of NuGet package analyzers come from packaages.
+
+Alternative designs are listed in the [Q & A](#Q%20&%20A) section below.
+
+### Source generator conflict resolution
+
+Source generators and analyzers are passed to the compiler as @(Analyzer) items.  These items may come from NuGet packages, be built into the SDK targets, be directly defined by the user via ProjectReference or raw Analyzer items, or be provided by the new framework mechanism described in this document.  All should be considered for conflict resolution based on the same constraint.
+
+The .NET SDK already does conflict resolution for other asset types: references and copy-local runtime assets.  It should do the same for source generators, ore more generally Analyzers.  Analyzers need not have "platform manifest" inputs for conflict resolution since all analyzers will be available during build, unlike runtime files which may not be available.  Comparison for the sake of conflict resolution can consider the same assembly and file version rules as reference files.  When finding two or more source generators with the same name conflict resolution will select the generator with higher assembly version, then higher file version, then that which is in the framework.  This parallels the same rules followed by reference assemblies.  The SDK may choose to optimize this comparison as it does for reference assemblies, by including a data file that includes analyzer metadata.
```

Comment:
```suggestion
The .NET SDK already does conflict resolution for other asset types: references and copy-local runtime assets.  It should do the same for source generators, or more generally Analyzers.  Analyzers need not have "platform manifest" inputs for conflict resolution since all analyzers will be available during build, unlike runtime files which may not be available.  Comparison for the sake of conflict resolution can consider the same assembly and file version rules as reference files.  When finding two or more source generators with the same name conflict resolution will select the generator with higher assembly version, then higher file version, then that which is in the framework.  This parallels the same rules followed by reference assemblies.  The SDK may choose to optimize this comparison as it does for reference assemblies, by including a data file that includes analyzer metadata.
```